### PR TITLE
Fix DocService to accept non JSON requests

### LINKED ISSUE #5218: DocService to accept non JSON requests
### Discussed in https://github.com/line/armeria/discussions/5214

<div type='discussions-op-text'>

<sup>Originally posted by **vthacker** September 30, 2023</sup>
Hello Armeria folks,

I'm using annotated doc service `@Post("/_bulk")` that can also parse NDJSON strings. This is the only dataformat that Elasticsearch/Opensearch accepts ( https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html )

The service works as expected, however I am running into an issue with the DocService

I am passing this as my HTTP Headers within the doc service 

```
{
  "content-type": "application/x-ndjson"
}
```

In the request body if I input NDJSON data, for example

```
{ "index" : { "_index" : "test", "_id" : "1" } }
{ "field1" : "value1" }

```

The doc serice parser is complaining since it's expecting data to be JSON
```
Error: Failed to parse a JSON object in the request body:
SyntaxError: Unexpected non-whitespace character after JSON at position 49 (line 2 column 1)
```

So my first questionn is there a way for me to disable the JSON parsing of the request body?

My second quesion is the DocService dseems to overwrite the content-type header. For example if I change my request body to be a valid JSON

```{ "index" : { "_index" : "test", "_id" : "1" } }```

and then "Copy as a curl command" I see this

```
curl -XPOST -H 'content-type: application/json; charset=utf-8' https:/service.com'/_bulk' -d '{ "index" : { "_index" : "test", "_id" : "1" } }
```

If I change the HTTP Header to be

```
{
  "Content-Type": "application/x-ndjson"
}
```

then I see the request being formed as 

```
curl -XPOST -H 'Content-Type: application/x-ndjson' -H 'content-type: application/json; charset=utf-8' https://service.com'/_bulk' -d '{ "index" : { "_index" : "test", "_id" : "1" } }
```
</div>

### PR DESCRIPTION
**Pull request by heowc**
Motivation:

1) Trying to use json pretty even though it's a string that can't use json pretty, like json-lines.
2) When copying as a curl command, it uses an array of characters to insert duplicate semantically identical headers.

Modifications:

1) Trying to use json pretty even though it's a string that can't use json pretty, like json-lines.
    - Fix some mime types to ignore pretty
2) When copying as a curl command, it uses an array of characters to insert duplicate semantically identical headers.
    - Change to `Headers` instead of string array

Result:

- Closes #5218

### CODE REVIEW

File: docs-client/src/lib/transports/transport.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ import { docServiceDebug, providers } from '../header-provider';
 
 import { Endpoint, Method } from '../specification';
 
+const ignoringPrettyMimeTypes: string[] = ['x-ndjson', 'json-seq'];
```
Comment (minwoox) [inline docs-client/src/lib/transports/transport.ts:22]: How about `nonJsonPrettifyMimeTypes`?

File: docs-client/src/lib/transports/transport.ts [hunk 2]
```diff
@@ -154,6 +159,14 @@ export default abstract class Transport {
     return {};
   }
 
+  private ignorePretty(applicationType: string) {
```
Comment (minwoox) [inline docs-client/src/lib/transports/transport.ts:162]: How about `shouldPrettify`?

**Review by minwoox — APPROVED** [review]
Apologize for the late review. 😓 
Looks good, thanks!

File: docs-client/src/lib/transports/transport.ts [hunk 3]
```diff
@@ -63,6 +65,9 @@ export default abstract class Transport {
     const applicationType = httpResponse.headers.get('content-type') || '';
     if (applicationType.indexOf('json') >= 0) {
       try {
+        if (this.ignorePretty(applicationType)) {
```
Comment (minwoox) [inline docs-client/src/lib/transports/transport.ts:68]: We should add this logic for the request body not for the response body. 😓 
https://github.com/line/armeria/blob/3af883e675201e0c119360bb786ac98b69db7246/docs-client/src/containers/MethodPage/DebugPage.tsx#L190

**Review by minwoox — CHANGES_REQUESTED** [review]

**Comment by jrhee17** [comment]
Pushed a small commit which

- Adds a `jsonLines` method for easier testing to `AnnotatedDocServiceTest`
- Extracts `validateJsonObject` so that multiple components can use it
- Check if `content-type` is overridden in `transport.ts` - If it is, then skip the json check

Let me know if changes don't make sense

**Review by minwoox — APPROVED** [review]
Thanks a lot! 😄

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

**Review by ikhoon — APPROVED** [review]
Thanks @heowc! 🙇‍♂️🙇‍♂️

**Comment by heowc** [comment]
Oops..!

I took a wrong approach, thank you for correcting me 🙇

**Pull request merged**